File: 09.palindrome-number/PalindromeNumber.go
package main

import "fmt"

func isPalindromeNumber(n int) bool {
	if n < 0 {
		return false
	}

	for n > 0 {

		if n < 10 {
			return true
		}

		i := 1
		left := n / 10
		for left > 10 {
			left /= 10
			i++
		}

		if left != n%10 {
			return false
		}

		for ; i > 0; i-- {
			left *= 10
		}

		n -= left
		n /= 10
	}

	return true
}

func solveProblem(n int) {
	fmt.Printf("%d is palindrome number: %v\n", n, isPalindromeNumber(n))
}

func main() {
	solveProblem(2)
	solveProblem(28273642)
	solveProblem(123456654321)
	solveProblem(17361)
}

File: 09.palindrome-number/PalindromeNumber.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

def is_palindrome_number(n: int) -> bool:
    if n < 0:
        return False

    while n > 0:
        if n < 10:
            return True

        i = 1
        left = n // 10
        while left > 10:
            left //= 10
            i += 1
        
        if left != n % 10:
            return False
        
        for _ in range(i):
            left *= 10

        n -= left
        n //= 10

    return True

def solve_problem(n : int):
	print(f"{n} is palindrome number: {is_palindrome_number(n)}")


if __name__ == "__main__":
	solve_problem(2)
	solve_problem(28273642)
	solve_problem(123456654321)
	solve_problem(17361)




